Themes (#318)
V4Fire/Client: src/super/i-static-page/modules/theme/README.md

# super/i-static-page/modules/theme

> This module provides a class and factory to theme managing.

Reviewer: ```suggestion
This module provides a class and factory to manage the app themes.
```

## Synopsis

> * Module uses several global variables from config:

Reviewer: ```suggestion
* The module uses several global variables from the config:
```

>   * `THEME` - initial theme name. If not specified, theme managing won't available.

Reviewer: ```suggestion
  * `THEME` - a name of the initial theme. If not specified, theme managing won't be available.
```

>   * `THEME_ATTRIBUTE` - attribute name to set an actual theme value on the root html element

Reviewer: ```suggestion
  * `THEME_ATTRIBUTE` - an attribute name to set a theme value to the root element
```

>   * `INCLUDED_THEMES` - a set of theme names, that available in the runtime

Reviewer: ```suggestion
  * `INCLUDED_THEMES` - a list of available app themes
```

> * Theme is set using property `current`

Reviewer: Тут надо синкнуть с замечаниями из кода

Author: Тут в итоге ок же с current?

* Available themes can be accessed by property `list`

## Events

| EventName      | Description            | Payload description | Payload                      |
| -------------- | ---------------------- | ------------------- | ---------------------------- |

> | `theme:change` | Theme value changed    | New and old value   | `string`; `CanUndef<string>` |

Reviewer: ```suggestion
| `theme:change` | Theme value has been changed    | The new and old value   | `string`; `CanUndef<string>` |
```
